### DISCUSSION TITLE
Video management in content type

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#10936] [Q&A]
How i upload videos in specific content type on admin side and later on show on differ pages by using content type.

**Comment by MikeAlhayek** [comment]
One way to do this is to create a part for the video (ex, `VideoPart`) with `MediaField`. then attach VideoPart to any content type you want.

If you want to customize how it renders, you can create a display driver for the VideoPart. I hope this helps

**Reply by saleemhaq** [reply]
can you please share the example if any? how i achieve end to end?

**Reply by MikeAlhayek** [reply]
You can do that using the UI.

But here is how you can do it via code. Note the code below is not tested.

In a `Models` folder add this class
```
public class VedioPart : ContentPart
{
    public MediaField File { get; set; }
}
```

In a `Migrations` folder add this class
```
public class MyCustomContentTypeMigrations : DataMigration
{
    private readonly IContentDefinitionManager _contentDefinitionManager;

    public EventMigrations(IContentDefinitionManager contentDefinitionManager) => _contentDefinitionManager = contentDefinitionManager;

    public int Create()
    {
        _contentDefinitionManager.AlterPartDefinition(nameof(VedioPart), part => part
            .Attachable()
            .WithField(nameof(VedioPart.Media), field => field
                .OfType(nameof(MediaField))
            )
        );

        _contentDefinitionManager.AlterTypeDefinition("MyCustomContentType", part => part
            .Listable()
            .Creatable()
            .Draftable()
            .Versionable()
            .Securable()
            .WithPart(nameof(TitlePart))
            .WithPart(nameof(VedioPart))
            // include other parts. you can also use the UI to modify the services.AddContentPart<VedioPart>();
services.AddScoped<IDataMigration, MyCustomContentTypeMigrations>();
        );

        return 1;
    }
}
```


Finally, in the start up class add these services
```
services.AddContentPart<VedioPart>();
services.AddScoped<IDataMigration, MyCustomContentTypeMigrations>();
```

Now, when you run the app, you should see a new content-type called `MyCustomContentType`. If you create a new content item, you should see a field that would allow you to upload media.
